src/pamiq_core/interactions/interval_adjustors.py

```diff
@@ -0,0 +1,63 @@
+import math
+from abc import ABC, abstractmethod
+
+from pamiq_core import time
+
+
+class BaseIntervalAdjustor(ABC):
+    """Adjusts the loop to run at a fixed interval."""
+
+    """Initial reset time is set to negative infinity for mathematical reasons."""
+
+    _last_reset_time: float = -math.inf
+
+    def __init__(self, interval: float, offset: float = 0.0) -> None:
+        """Constructs the IntervalAdjustor.
+
+        Args:
+            interval: The desired time between each invocation in seconds.
+            offset: The initial time offset to subtract from the interval on the first loop iteration,
+                allowing adjustment of the actual interval for each computer using this parameter.
+        """
+        self._interval = interval
+        self._offset = offset
+        self._time_to_wait = interval - offset
+
+    def reset(self) -> float:
+        """Resets the start time of this adjustor to the current time.
+
+        Returns:
+            float: The start time after resetting the timer, as a high precision time counter.
+        """
+        self._last_reset_time = time.perf_counter()
+        return self._last_reset_time
+
+    @abstractmethod
+    def _adjust(self) -> None:
```

Comment:
コーディング規約的に `adjust_impl` にするべきだった。
`_` prefixはpythonにおけるprivateを慣習的に表すので、サブクラスでoverrideされることを前提とする抽象メソッドに対しては使うべきではないんだ
```suggestion
    def adjust_impl(self) -> None:
```